tools/clang/tools/dxcvalidator/dxcvalidator.cpp

```diff
@@ -31,6 +32,42 @@
 using namespace llvm;
 using namespace hlsl;
 
+static void HashAndUpdate(DxilContainerHeader *Container) {
+  // Compute hash and update stored hash.
+  // Hash the container from this offset to the end.
+  static const uint32_t DXBCHashStartOffset =
+      offsetof(struct DxilContainerHeader, Version);
+  const unsigned char *DataToHash =
+      (const unsigned char *)Container + DXBCHashStartOffset;
+  unsigned AmountToHash = Container->ContainerSizeInBytes - DXBCHashStartOffset;
+  ComputeHashRetail(DataToHash, AmountToHash, Container->Hash.Digest);
+}
+
+static void HashAndUpdateOrCopy(uint32_t Flags, IDxcBlob *Shader,
+                                IDxcBlob **Hashed) {
+  if (Flags & DxcValidatorFlags_InPlaceEdit) {
+    HashAndUpdate((DxilContainerHeader *)Shader->GetBufferPointer());
+    *Hashed = Shader;
+    Shader->AddRef();
+  } else {
+    CComPtr<AbstractMemoryStream> HashedBlobStream;
+    uint32_t HR =
+        CreateMemoryStream(DxcGetThreadMallocNoRef(), &HashedBlobStream);
+    if (FAILED(HR))
+      throw hlsl::Exception(HR);
```

Comment:
Just a note that you could have used the `IFT()` macro here and below since the return value isn't used later.

Reply:
There's very much an existing style in this file to use `IFT` here.  If we're going to continue to use exception based error handling in here then I think we should follow the convention that's here already.

If we're going to change the way error handling is done in this file then we might want to have a conversation about using exceptions at all here.

Reply:
Updated.